src/status_im/contexts/communities/actions/addresses_for_permissions/view.cljs

```diff
@@ -1,55 +1,76 @@
-        accounts                    (rf/sub [:wallet/accounts-with-customization-color])]
-    [rn/safe-area-view {:style style/container}
+        accounts                    (rf/sub [:wallet/accounts-with-customization-color])
+        selected-addresses          (reagent/atom (rf/sub [:communities/addresses-for-permissions]))]
+    (fn []
+      [rn/safe-area-view {:style style/container}
+       [quo/drawer-top
+        {:type                :context-tag
+         :title               (i18n/label :t/addresses-for-permissions)
+         :community-name      name
+         :button-icon         :i/info
+         :on-button-press     not-implemented/alert
+         :community-logo      (get-in images [:thumbnail :uri])
+         :customization-color color}]
 
-     [quo/drawer-top
-      {:type                :context-tag
-       :title               (i18n/label :t/addresses-for-permissions)
-       :community-name      name
-       :button-icon         :i/info
-       :on-button-press     not-implemented/alert
-       :community-logo      (get-in images [:thumbnail :uri])
-       :customization-color color}]
+       [rn/flat-list
+        {:render-fn               account-item
+         :render-data             selected-addresses
+         :content-container-style {:padding 20}
+         :key-fn                  :address
+         :data                    accounts}]
 
-     [rn/flat-list
-      {:render-fn               account-item
-       :content-container-style {:padding 20}
-       :key-fn                  :address
-       :data                    accounts}]
+       (when (empty? @selected-addresses)
+         [rn/view
+          {:style style/error-message}
+          [quo/icon
+           :i/info
+           {:color colors/danger-50
+            :size  16}]
+          [quo/text
+           {:size  :paragraph-2
+            :style {:color colors/danger-50}}
+           (i18n/label :t/no-addresses-selected)]])
 
-     [rn/view {:style style/buttons}
-      [quo/button
-       {:type            :grey
-        :container-style {:flex 1}
-        :on-press        #(rf/dispatch [:navigate-back])}
-       (i18n/label :t/cancel)]
-      [quo/button
-       {:container-style     {:flex 1}
-        :customization-color color
-        :on-press            #(rf/dispatch [:navigate-back])}
-       (i18n/label :t/confirm-changes)]]]))
+       [rn/view {:style style/buttons}
+        [quo/button
+         {:type            :grey
+          :container-style {:flex 1}
+          :on-press        #(rf/dispatch [:navigate-back])}
+         (i18n/label :t/cancel)]
+        [quo/button
+         {:container-style     {:flex 1}
+          :customization-color color
+          :disabled?           (empty? @selected-addresses)
+          :on-press            (fn []
+                                 (rf/dispatch [:communities/set-addresses-for-permissions
+                                               @selected-addresses])
```

Comment:
Similar here, the selected addresses state should be in the app db, so there would be no need to pass it around when dispatching events. Moving the state to the app db would also allow us to cover more of the logic with unit tests for event handlers.